modules/transports/core/nhttp/src/main/java/org/apache/synapse/transport/netty/sender/SourceResponseHandler.java

```diff
@@ -419,6 +436,34 @@ public static void sendResponse(MessageContext msgCtx, HttpCarbonMessage inbound
         serializeData(msgCtx, outboundResponseMsg);
     }
 
+    /**
+     * Send the server push promise to the client
+     *
+     * @param http2PushPromise Http2PushPromise
+     * @param clientRequest    HttpCarbonMessage
+     * @throws AxisFault if error occurred while pushing responses.
+     */
+    public static void pushPromise(Http2PushPromise http2PushPromise, HttpCarbonMessage clientRequest) throws AxisFault {
```

Comment:
Method comment should end with a period, the variables and returned types need to be explained. shall we check and fix other places as well.